recipes/polymorphic_value/all/test_package/conanfile.py

```diff
@@ -1,16 +1,21 @@
-from conans import ConanFile, CMake, tools
-import os
+from conan import ConanFile
+from conan.tools.build import can_run
+from conan.tools.cmake import CMake
 
 
 class TestPackageConan(ConanFile):
     settings = "os", "arch", "compiler", "build_type"
-    generators = "cmake", "cmake_find_package_multi"
+    generators = "CMakeDeps", "CMakeToolchain"
+
+    def requirements(self):
+        self.requires(self.tested_reference_str)
 
     def build(self):
         cmake = CMake(self)
         cmake.configure()
         cmake.build()
 
     def test(self):
-        if not tools.cross_building(self):
-            self.run(os.path.join("bin", "test_package"), run_environment=True)
+        if can_run(self):
+            cmake = CMake(self)
+            cmake.test()
```

Comment:
The correct would run `self.run` which I prefer because you need to invoke with `env=conanrun` and I'm not sure if `cmake.test()` preserves it. So please, follow https://github.com/conan-io/conan-center-index/blob/master/docs/package_templates/cmake_package/all/test_package/conanfile.py as example which we know is safe.